compaction/compaction_manager.cc

```diff
@@ -326,11 +326,11 @@ future<sstables::compaction_result> compaction_manager::task::compact_sstables(s
     descriptor.replacer = [this, &t, release_exhausted] (sstables::compaction_completion_desc desc) {
         t.get_compaction_strategy().notify_completion(desc.old_sstables, desc.new_sstables);
         _cm.propagate_replacement(t.as_table_state(), desc.old_sstables, desc.new_sstables);
-        t.on_compaction_completion(desc);
         // Calls compaction manager's task for this compaction to release reference to exhausted SSTables.
         if (release_exhausted) {
             release_exhausted(desc.old_sstables);
         }
+        t.on_compaction_completion(std::move(desc));
```

Comment:
Can this change in order cause problems?

Reply:
It's good practice to make these refactoring have as little effect as possible, and push out more dangerous changes out.